### ISSUE TITLE
Default user ka-lite on Ubuntu Software Center 15.04

### ISSUE THREAD

**Issue by radinamatic** [learningequality/installers#212]
Default user ka-lite on Ubuntu

**Installer**: ka-lite-bundle 0.15~b3-0ubuntu1, through the **Ubuntu Software Center**  

**Expected Behavior**: What I was eager to see were the two screenshots from our [docs on Linux installation](http://ka-lite.readthedocs.org/en/0.15.x/installguide/install_all.html#linux), but no luck apparently.

**Current Behavior**: After a slight chaos I had going on described in [#4549](https://github.com/learningequality/ka-lite/issues/4549), I tried to do a completely clean install:

- first I used the soft removal `sudo apt-get remove ka-lite-gtk ka-lite ka-lite-bundle` 
- then the hard one `sudo apt-get purge ka-lite-gtk ka-lite ka-lite-bundle` 
- deleted `~/.kalite` folder 
- deleted `/etc/ka-lite/ ` folder
- deleted `var/ka-lite/.kalite/` folder I had no idea existed in the first place, but seemed to be the cause of my **ka-lite-gtk** pains in [#4549](https://github.com/learningequality/ka-lite/issues/4549).
- used the `ps ax|grep kalite` to see if there were any processes left running prior to new install 

*Note: it would be helpful to know if there are more steps to do, in order to perform a clean install, a part from the ones above.*
 
After I started the installation in the **Ubuntu Software Center**, only 2 **debconf** windows appeared: 

![startup1](https://cloud.githubusercontent.com/assets/1457929/10312802/f8510340-6c4c-11e5-93ba-d7b5da9b5d90.png)

![startup2](https://cloud.githubusercontent.com/assets/1457929/10312923/764ce62e-6c4d-11e5-82aa-1864cfacc525.png)

##### There were 3 moments during the installation when some window seemed to appear for a split second, but not enough for me to see what was in them, not to talk about making a selection or a conscious decision.  

After the installation finished, **Ubuntu Software Center** informed me that I should run the command `kalite`, which I did and got the following reply (@EdDixon described something similar in #145):

![default-user-ka-lite-ubuntu](https://cloud.githubusercontent.com/assets/1457929/10312932/7f2f43ea-6c4d-11e5-8121-f321cc9c890c.png)

Now, it may be just me, but I find this message frustratingly unclear... :( 
```
didi@didi-Inspiron-3531:~$ kalite
You are not running kalite as the default user ka-lite. 
```

After this I'm left wondering *when the heck was this mysterious* **default ka-lite** *user even created*? The whole installation and running the command `kalite` was done while I was logged in as **my own** default user `didi`, and at no point had I consciously consented to the creation of another user. Are those 3 windows that did not give me a chance to see them the one I'm missing from from our [docs on Linux installation](http://ka-lite.readthedocs.org/en/0.15.x/installguide/install_all.html#linux) about the creation of this **default ka-lite** user? 

I get the same message if I try to run `kalite manage setup` command in the Terminal. Maybe all this would sound perfectly logical to somebody with more system-service-command-line experience, but leaves me scratching my head how to proceed (y/No).    

So I see two TODOs here: 
1. Find out why am I not getting the default user selection question
2. Give more consistent message to the users: are they to run `kalite manage setup` (as in **debconf** window), or only the `kalite` one (as per final **Ubuntu Software Center** window).

**Comment by benjaoming** [comment]
@radinamatic you never see the screen shown in #146 ?

**Comment by radinamatic** [comment]
Unfortunately not.

**Comment by benjaoming** [comment]
Grr, that Software Center should abide debconf, not sure why it doesn't.

What version of Ubuntu is this? 14.04? 14.10? 15.04?

**Comment by radinamatic** [comment]
Sorry, should have noted that at the beginning: 15.04 with all the latest updates.

**Comment by radinamatic** [comment]
@benjaoming I think I may have figured out something that can lead us out of the woods:

Could this have to do something with the fact that I **did not** actually choose to run on startup? The first screenshot above displays the checkbox as if I had, I apologize for that, but from the second it's clear that the choice was NOT to install system service. 

I reinstalled yesterday with the the `dpkg -i` command, and if I remember correctly, the sequence was this:
```
didi@didi-Inspiron-3531:~/Desktop$ sudo dpkg -i ka-lite-bundle_0.15~b3-0ubuntu1_all.deb
Selecting previously unselected package ka-lite-bundle.
(Reading database ... 367131 files and directories currently installed.)
Preparing to unpack ka-lite-bundle_0.15~b3-0ubuntu1_all.deb ...
Unpacking ka-lite-bundle (0.15~b3-0ubuntu1) ...
Setting up ka-lite-bundle (0.15~b3-0ubuntu1) ...
```
![startup1-dpkg](https://cloud.githubusercontent.com/assets/1457929/10338423/467499d8-6d06-11e5-9da1-ac9d07bc3035.png)

**Again, I chose not to run at strartup**

![startup2-dpkg](https://cloud.githubusercontent.com/assets/1457929/10338442/680ea2e6-6d06-11e5-83c3-51d1f96e22a9.png)
 
```
...
Setting up ka-lite-bundle (0.15~b3-0ubuntu1) ...
Setting ownership to didi for assessment item database
Copying init.d script

Creating config file /etc/init.d/ka-lite with new version
didi@didi-Inspiron-3531:~/Desktop$ kalite manage setup
...
```

From the output above, one can conclude that in this case my own user was the designated owner, but I still did not get any screen like the one you linked in #146. So this difference might point to an debconf issue with Ubuntu Software Center, that makes the **ka-lite** as the default user/owner **without asking permission** from the user when they do not choose to run at startup. On the other hand `dpkg -i` command **also** does not ask for user's permission, but at least it sets up their own user as the owner, which is the *sane default*.

Then I re-read more carefully our docs on Linux installation, and rightfully, the second step said:

> **If you chose to start on boot** in the previous step, you will be prompted to choose the owner for the KA Lite server process.

To test the proposition, I uninstalled and purged everything again, restarted the installation with Ubuntu Software Center, choose to run on startup, and low and behold, I finally got to see that screen:

![startup2-default-user](https://cloud.githubusercontent.com/assets/1457929/10340954/d22276ba-6d12-11e5-97e2-9e8ef0e8dddc.png)
 
Ok, so to draw some conclusion from all of this, I'd say that it's necessary to separate the two things (startup and default user selection), and offer the users option to choose the default owner **weather** they prefer to run on startup or not. This to me seems more respectful to users and gives them more control, and saves us dealing with this mess in the future. What do you think?

**Comment by benjaoming** [comment]
Thanks for investigating!

Just repeating this for the sake of ensuring it isn't a factor: To clear settings, you need to *purge* rather than *remove* which keeps the debconf from previous install.

I think the conclusion has to be that **no matter if a user chooses to run the system service or not, she should still be asked about the contents of /etc/ka-lite/username**

Just remember that no matter what we do during installation, this can always be changed in a comfortable UI afterwards, ka-lite-gtk :) I'd much rather that this works 100% and we start pushing that solution to extensively supplement the debconf.

**Comment by radinamatic** [comment]
> Thanks for investigating!

Love the feeling of finding culprits! ;)

> Just repeating this for the sake of ensuring it isn't a factor: To clear settings, you need to purge rather than remove which keeps the debconf from previous install.

I usually do both, first `remove`, then `purge`, talk about OCD... 


> I think the conclusion has to be that no matter if a user chooses to run the system service or not, she should still be asked about the contents of /etc/ka-lite/username

To `run the system service` means running KA LIte server at startup, in the background, right?  


> Just remember that no matter what we do during installation, this can always be changed in a comfortable UI afterwards, ka-lite-gtk :) I'd much rather that this works 100% and we start pushing that solution to extensively supplement the debconf.

Well, now that I have my own user `didi` as the default, everything seems to be running fine. 

![ka lite control panel_details](https://cloud.githubusercontent.com/assets/1457929/10369008/4f004eee-6dd8-11e5-9dc7-ba0ddc4a5701.png)

The only question I have left (for now) is the location of the assessment database which in my case (above) is `/usr/share/kalite/assessment/khan/assessmentitems.sqlite`. Should we include that in the list of files to purge prior to doing a new, completely clean install (on Trello, for our own testing purposes)?

**Comment by benjaoming** [comment]
> The only question I have left (for now) is the location of the assessment database which in my case (above) is `/usr/share/kalite/assessment/khan/assessmentitems.sqlite`. Should we include that in the list of files to purge prior to doing a new, completely clean install (on Trello, for our own testing purposes)?

That location is populated during the installation of the deb, by the debian scripts. They are made sensitive to whether there is data in the location or not.

I have to ponder if upgrading the package will cause problems.  Because upgrading from 0.14.0 to 0.14.1 shouldn't re-download anything, while upgrading to 0.15 should force a new download.

Hm. Annoying. Assessment items just shouldn't be part of the software distribution :(

**Comment by radinamatic** [comment]
> Assessment items just shouldn't be part of the software distribution :(

I don't know what is reasoning behind this location `/usr/share/kalite/` for assessment items and their DB. Since that is, essentially, a type of content, and especially in the multichannel scenario, it makes total sense to me for it to be separated from the software itself.

![files](https://cloud.githubusercontent.com/assets/1457929/10372987/0221f3da-6deb-11e5-8fda-d505e5a7dfdd.png)

**Comment by benjaoming** [comment]
> I don't know what is reasoning behind this location /usr/share/kalite/ for assessment items and their DB.

Here's the thing: You include it in the software, and you have to install it somewhere. During installation on a multi-user system, you won't know which use to install it for. So it has to go in a system location.

Screenshot is for? Is it a very complicated way of showing `ls -l /usr/share/kalite`? ;)

**Comment by radinamatic** [comment]
> During installation on a multi-user system, you won't know which use to install it for. So it has to go in a system location.

Ok, understood, but why do videos & subs deserve a place in the home folder (which means that you are picking one specific user on the multi-user system) and assessments do not? I'm not saying either way, I just do not get the difference...

> Screenshot is for?

Nah, I'm a sucker for [pics](http://radinamatic.com/blog/?page_id=543)...

**Comment by benjaoming** [comment]
In your menu...

```
show
me:

    ALL
    LANDSCAPE
    People
    Sky
    Volcano
    Art
    HDR
    B&W
    Architecture
    Life
```

You should add "Screenshots of ka-lite-gtk" :D

**Comment by benjaoming** [comment]
> Ok, understood, but why do videos & subs deserve a place in the home folder (which means that you are picking one specific user on the multi-user system) and assessments do not? I'm not saying either way, I just do not get the difference...

Because videos and subs are NEVER distributed in the app. If you download assessment items through `kalite manage setup` you will also have them placed in the home folder.

In case we were to distribute other contents with the debian pkg, it would also have to go in /usr/share/kalite as a system-wide concept.

**Comment by radinamatic** [comment]
If we count all the screenshots I made for KA Lite, GTK doesn't even get close to the first place. :stuck_out_tongue: 

> If you download assessment items through "kalite manage setup" you will also have them placed in the home folder.

Ha, tried that once, took three eternities, never again... Good to know that's how it's supposed to work.
 
> In case we were to distribute other contents with the debian pkg, it would also have to go in /usr/share/kalite as a system-wide concept.

Ok, so to return to my previous question:assessment/khan/assessmentitems.sqlite. Should we include the ` /usr/share/kalite/` in the list of things to purge prior to doing a new, completely clean install?

**Comment by benjaoming** [comment]
> Ok, so to return to my previous question:assessment/khan/assessmentitems.sqlite. Should we include the /usr/share/kalite/ in the list of things to purge prior to doing a new, completely clean install?

No not IMO. Since all files distributed by ka-lite og ka-lite-bundle are managed by the packaging system, it will be removed when running `apg-get remove` or overwritten if upgrading a package.

**Comment by radinamatic** [comment]
Ok... :+1:

**Comment by benjaoming** [comment]
@radinamatic could you give a summary of what the issue here was?

**Comment by radinamatic** [comment]
When user installs KA Lite .deb through Ubuntu Software Center (UBS), and chooses **NOT to run on startup** (unchecks the box in the first screenshot above), UBS (or `debconf` or the installer?) creates the user **ka-lite** and makes it the default KA Lite user (set in `/etc/ka-lite/` folder) **without asking user's permission**. If the users leaves the checkbox as-is, UBS does offer them the option to choose which user will be the owner of KA Lite.

When user installs with `dpkg -i` command and choose not to run on startup, they equally do not get asked the permission which user they want to be the owner, but at least that command sets up their own user (not the `ka-lite` one as UBS) as the owner.

**Comment by benjaoming** [comment]
Sorry, there's a lot of content here. But the title of the Issue was fixed in #146
